src/dataunion/DataUnion.ts

```diff
@@ -564,30 +590,109 @@ export class DataUnion {
         return new Contracts(this.client)
     }
 
-    // template for withdraw functions
-    // client could be replaced with AMB (mainnet and sidechain)
+    /**
+     * Template for withdraw functions
+     * @private
+     * @returns the sidechain withdraw transaction receipt IF called with sendToMainnet=false,
+     *          ELSE the message hash IF called with payForTransport=false and waitUntilTransportIsComplete=false,
+     *          ELSE the mainnet AMB signature execution transaction receipt IF we did the transport ourselves,
+     *          ELSE null IF transport to mainnet was done by someone else (in which case the receipt is lost)
+     */
     private async _executeWithdraw(
-        getWithdrawTxFunc: () => Promise<TransactionResponse>,
+        getWithdrawTxFunc: () => Promise<ContractTransaction>,
         recipientAddress: EthereumAddress,
         options: DataUnionWithdrawOptions = {}
-    ): Promise<TransactionReceipt> {
+    ): Promise<ContractReceipt | AmbMessageHash | null> {
         const {
             pollingIntervalMs = 1000,
-            retryTimeoutMs = 60000,
-            freeWithdraw = this.client.options.dataUnion.freeWithdraw,
+            retryTimeoutMs = 300000,
+            // by default, transport the signatures if freeWithdraw isn't supported by the sidechain
+            payForTransport = !this.client.options.dataUnion.freeWithdraw,
```

Comment:
Is `payForTransport` just the same concept as `freeWithdraw`, but negated? Do we need both terms?

Reply:
it is. One is the property of the bridge, one is the property of the function call, it's like the opposite angle of looking at it.

But I'll still change it so that the name is the same. So freeWithdraw -> !payForTransport everywhere. It's better that way.